core/src/main/java/org/pentaho/di/core/row/ValueDataUtil.java

```diff
@@ -383,30 +427,62 @@ public static String createChecksum( ValueMetaInterface metaA, Object dataA, Str
 
       md5Hash = md5HashBuff.toString();
 
+    } catch ( KettleFileNotFoundException e ) {
+      if ( failIfNoFile ) {
+        throw e;
+      }
     } catch ( Exception e ) {
-      // ignore - should likely log the exception
+      log.debug( e.getMessage() );
     } finally {
-      try {
-        if ( in != null ) {
-          in.close();
-        }
-      } catch ( Exception e ) {
-        // Ignore
-      }
+      IOUtils.closeQuietly( file );
     }
-
     return md5Hash;
   }
 
+  /**
+   * @deprecated Use {@link ValueDataUtil#checksumCRC32(ValueMetaInterface, Object, boolean)} instead
+   */
+  @Deprecated
   public static Long ChecksumCRC32( ValueMetaInterface metaA, Object dataA ) {
     long checksum = 0;
+    try {
+      checksum = checksumCRC32( metaA, dataA, false );
+    } catch ( KettleFileNotFoundException e ) {
+      // Ignore
+    }
+    return checksum;
+  }
+
+  /**
+   *
+   * @param metaA
+   *   The ValueMetaInterface
+   * @param dataA
+   *   Filename
+   * @param failIfNoFile
+   *   Indicates if the transformation should fail if no file is found
+   * @return File's CRC32 checksum
+   * @throws KettleFileNotFoundException
+   */
+  public static Long checksumCRC32( ValueMetaInterface metaA, Object dataA, boolean failIfNoFile )
+          throws KettleFileNotFoundException {
+    long checksum = 0;
+
+    if ( dataA == null ) {
```

Comment:
This actually is not a file not found.... it's a file path/name not provided.... I'm not sure the file not found exception suits this case